Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1321,83 +1321,246 @@ auto RenderTheme::colorCache(OptionSet<StyleColorOptions> options) const -> Colo
+    // Background of application content or documents.
+    case CSSValueCanvas:
+        return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-canvastext
+    // Text in application content or documents.
+    case CSSValueCanvastext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-linktext
+    // Text in non-active, non-visited links. For light backgrounds, traditionally blue.
     case CSSValueLinktext:
         return SRGBA<uint8_t> { 0, 0, 238 };
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-visitedtext
+    // Text in visited links. For light backgrounds, traditionally purple.
     case CSSValueVisitedtext:
         return SRGBA<uint8_t> { 85, 26, 139 };
-    case CSSValueActiveborder:
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-activetext
+    // Text in active links. For light backgrounds, traditionally red.
+    case CSSValueActivetext:
+    case CSSValueWebkitActivelink: // Non-standard addition.
+        return Color::red;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonface
+    // The face background color for push buttons.
+    case CSSValueButtonface:
+        return Color::lightGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttontext
+    // Text on push buttons.
+    case CSSValueButtontext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonborder
+    // The base border color for push buttons.
+    case CSSValueButtonborder:
+        return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-field
+    // Background of input fields.
+    case CSSValueField:
         return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-fieldtext
+    // Text in input fields.
+    case CSSValueFieldtext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-highlight
+    // Background of selected text, for example from ::selection.
+    case CSSValueHighlight:
+        return SRGBA<uint8_t> { 181, 213, 255 };
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-highlighttext
+    // Text of selected text.
+    case CSSValueHighlighttext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-selecteditem
+    // Background of selected items, for example a selected checkbox.
+    case CSSValueSelecteditem:
+        return Color::lightGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-selecteditemtext
+    // Text of selected items.
+    case CSSValueSelecteditemtext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-mark
+    // Background of text that has been specially marked (such as by the HTML mark element).
+    case CSSValueMark:
+        return Color::yellow;
```

Comment:
Given the existence of `RenderTheme::platformAnnotationHighlightColor` I am not sure what this should be returning.

Reply:
I think we should return `platformAnnotationHighlightColor` here.

Reply:
That would be a change in behavior for <mark>, so I'd prefer to leave that to a subsequent change.